tests/vcs/mercurial/HgCommitBearTest.py

```diff
@@ -0,0 +1,504 @@
+        self.assertEqual(self.run_uut(body_close_issue=True,),
+                         ['Invalid issue number: #01112'])
+        self.assert_no_msgs()
+
+        # Bitbucket host with no full issue reference
+        self.hg_commit('Shortlog\n\n'
+                       'First line, blablablablablabla.\n'
+                       'Another line, blablablablablabla.\n'
+                       'Fix #1112')
+        self.assertEqual(self.run_uut(
+                             body_close_issue=True,
+                             body_close_issue_full_url=True),
+                         ['Host bitbucket does not support full issue '
+                          'reference.'])
+        self.assert_no_msgs()
+
+        # Invalid characters in issue number
+        self.hg_commit('Shortlog\n\n'
+                       'First line, blablablablablabla.\n'
+                       'Another line, blablablablablabla.\n'
+                       'Fix #1112-3')
+        self.assertEqual(self.run_uut(
+                             body_close_issue=True),
+                         ['Invalid issue reference: #1112-3'])
+        self.assert_no_msgs()
+
+        # One of the short references is broken
+        self.hg_commit('Shortlog\n\n'
+                       'First line, blablablablablabla.\n'
+                       'Another line, blablablablablabla.\n'
+                       'Resolve #11 and closes #notnum')
+        self.assertEqual(self.run_uut(body_close_issue=True),
+                         ['Invalid issue reference: #notnum'])
+        self.assert_no_msgs()
+
+        # Incorrect close issue other repo pattern
+        self.hg_commit('Shortlog\n\n'
+                       'First line, blablablablablabla.\n'
+                       'Another line, blablablablablabla.\n'
+                       'Fix #11 and close bitbucket#32')
+        self.assertEqual(self.run_uut(body_close_issue=True),
+                         ['Invalid issue reference: bitbucket#32'])
+        self.assert_no_msgs()
+
+    def test_different_path(self):
+        no_hg_dir = mkdtemp()
+        self.hg_commit('Add a very long shortlog for a bad project history.')
+        os.chdir(no_hg_dir)
+        # When section doesn't have a project_dir
+        self.assertEqual(self.run_uut(), [])
+        hg_error = self.msg_queue.get().message
+        self.assertEqual(hg_error[:4], 'hg: ')
+        # when section does have a project_dir
+        self.section.append(Setting('project_dir', escape(self.hgdir, '\\')))
+        self.assertEqual(self.run_uut(),
+                         ['Shortlog of the HEAD commit contains 51 '
+                          'character(s). This is 1 character(s) longer than '
+                          'the limit (51 > 50).'])
+        self.assertEqual(get_config_directory(self.section),
+                         self.hgdir)
+        os.chdir(self.hgdir)
+        os.rmdir(no_hg_dir)
+
+    def test_nltk_download_disabled(self):
```

Comment:
These nltk tests also belong in a new `CommitBearTest`
create a FakeCommitBear in the test module if you need to.